src/swarm/neo/connection/RequestOnConnBase.d

```diff
@@ -411,9 +411,27 @@ abstract class RequestOnConnBase
             *******************************************************************/
 
             public void add ( T ) ( ref T elem )
+            {
+                this.addPointer(&elem);
+            }
+
+            /*******************************************************************
+
+                Adds a single element to the payload to be sent.
+
+                Note that this takes a pointer to the element, and the element's
+                lifetime needs to match or exceed that of this payload to avoid
+                memory corruption.
+
+                Params:
+                    ptr = pointer to the element to add
+
+            *******************************************************************/
+
+            public void addPointer ( T ) ( const T* elem )
```

Comment:
Ideally it should override the function `add`  with different input parameter.

Reply:
I think we should deprecate / remove add in the future. It exposes too much a dangerous interface.
This way we'll just have `addCopy` / `addPointer` / `addArray` (which calls `addCopy`).
Perhaps a more descriptive name can be used, such as `addScopedData` ?

Reply:
Then it's fine to keep it as it is and no need to change the name.